### PR TITLE
"move type bounds to where clause" assist

### PR DESCRIPTION
**Pull request by viorina**

### CODE REVIEW

File: crates/ra_assists/src/move_bounds.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            let params_with_bounds = type_param_list
+                .type_params()
+                .filter(|p| p.type_bound_list().is_some())
+                .collect::<Vec<_>>();
+
+            for param in &params_with_bounds {
+                let colon =
+                    param.syntax().children_with_tokens().find(|it| it.kind() == COLON).unwrap();
+                let start = colon.text_range().start();
+                let end = param.type_bound_list().unwrap().syntax().text_range().end();
```
Comment (matklad) [inline crates/ra_assists/src/move_bounds.rs]: This might fire if there's a param without a bound?

Comment (viorina) [reply crates/ra_assists/src/move_bounds.rs]: No, because we iterate the parameters with a bound only.

Comment (matklad) [reply crates/ra_assists/src/move_bounds.rs]: Ah, riiight....

Let's add a test for this though: in the existing tests, all parameters have bounds.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**